cmake/gen_version_h.cmake

```diff
@@ -16,6 +16,10 @@ if(NOT DEFINED ${BUILD_VERSION_NAME})
   git_describe(${work_dir} ${BUILD_VERSION_NAME})
 endif()
 
+git_commit_hash(${work_dir} GIT_COMMIT_HASH)
+git_commit_hash_short("${GIT_COMMIT_HASH}" GIT_COMMIT_HASH_SHORT GIT_COMMIT_HASH_SHORT_UINT32)
```

Comment:
this call will fail the script, and thereby also fail the build if `git` was not found or returned something which cannot be parsed.

Please ensure a valid `version.h` is still generated, even if `git` is missing.

There is a reason why existing code has this:
https://github.com/zephyrproject-rtos/zephyr/blob/cc544803811bd91c2500570204eb5fe2aeeea6f6/cmake/modules/git.cmake#L15